tests/EventLoopTest/CancellationTest.php

```diff
@@ -0,0 +1,261 @@
+<?php
+
+namespace M6WebTest\Tornado\EventLoopTest;
+
+use M6Web\Tornado\CancelledException;
+use M6Web\Tornado\EventLoop;
+
+trait CancellationTest
+{
+    abstract protected function createEventLoop(): EventLoop;
+
+    public function testCanceledPromiseThrowException()
+    {
+        $eventLoop = $this->createEventLoop();
+        $promise = $eventLoop->delay(self::LONG_WAITING_TIME);
+        $promise->cancel();
+
+        $this->expectException(CancelledException::class);
+        $eventLoop->wait($promise);
+    }
+
+    public function testCannotCancelFulfilledPromise()
+    {
+        $eventLoop = $this->createEventLoop();
+        $promise = $eventLoop->promiseFulfilled(42);
+        $promise->cancel();
+
+        $this->assertSame(42, $eventLoop->wait($promise));
+    }
+
+    public function testCannotCancelRejectedPromise()
+    {
+        $eventLoop = $this->createEventLoop();
+        $expectedException = new class() extends \Exception {
+        };
+
+        $promise = $eventLoop->promiseRejected($expectedException);
+        $promise->cancel();
+
+        $this->expectException(get_class($expectedException));
+        $eventLoop->wait($promise);
+    }
+
+    public function testCancellationDoesNotInterruptExecution()
+    {
+        $eventLoop = $this->createEventLoop();
+        $promise = $eventLoop->idle();
+
+        $result = $eventLoop->wait($eventLoop->async((function () use ($promise) {
+            yield $promise;
+            $promise->cancel();
+
+            return 'success';
+        })()));
+        $this->assertSame('success', $result);
+    }
+
+    public function testCancellationThrowAnCancelledException()
```

Comment:
Looks similar to `testCanceledPromiseThrowException` 🤔 
If we intend to do the test for `idle` and `delay`, it should be explicit in test name, something like `testIdleThrowACancelledException` ?